streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/TaskMovement.java

```diff
@@ -132,22 +132,97 @@ static boolean assignTaskMovements(final Map<TaskId, SortedSet<UUID>> tasksToCau
         return movementsNeeded;
     }
 
+    static int assignStandbyTaskMovements(final Map<TaskId, SortedSet<UUID>> tasksToCaughtUpClients,
```

Comment:
This algorithm is similar to the active one, but there are also important differences, so I didn't converge them.